receiver/postgresqlreceiver/client.go

```diff
@@ -17,24 +17,31 @@ package postgresqlreceiver // import "github.com/open-telemetry/opentelemetry-co
 import (
 	"context"
 	"database/sql"
-	"errors"
 	"fmt"
 	"net"
 	"strings"
 
 	"github.com/lib/pq"
 	"go.opentelemetry.io/collector/config/confignet"
 	"go.opentelemetry.io/collector/config/configtls"
-	"go.opentelemetry.io/collector/receiver/scrapererror"
+	"go.uber.org/multierr"
 )
 
+// databaseName is a name that refers to a database so that it can be uniquely referred to later
+// i.e. database1
+type databaseName string
+
+// tableIdentifier is an identifier that contains both the database and table separated by a "|"
+// i.e. database1|table2
+type tableIdentifier string
+
 type client interface {
 	Close() error
-	getCommitsAndRollbacks(ctx context.Context, databases []string) ([]MetricStat, error)
-	getBackends(ctx context.Context, databases []string) ([]MetricStat, error)
-	getDatabaseSize(ctx context.Context, databases []string) ([]MetricStat, error)
-	getDatabaseTableMetrics(ctx context.Context) ([]MetricStat, error)
-	getBlocksReadByTable(ctx context.Context) ([]MetricStat, error)
+	getDatabaseStats(ctx context.Context, databases []string) (map[databaseName]databaseStats, error)
+	getBackends(ctx context.Context, databases []string) (map[databaseName]int64, error)
+	getDatabaseSize(ctx context.Context, databases []string) (map[databaseName]int64, error)
+	getDatabaseTableMetrics(ctx context.Context, db string) (map[tableIdentifier]tableStats, error)
+	getBlocksReadByTable(ctx context.Context, db string) (map[tableIdentifier]tableIOStats, error)
```

Comment:
```suggestion
	getDatabaseStats(ctx context.Context, databases []databaseName) (map[databaseName]databaseStats, error)
	getBackends(ctx context.Context, databases []databaseName) (map[databaseName]int64, error)
	getDatabaseSize(ctx context.Context, databases []databaseName) (map[databaseName]int64, error)
	getDatabaseTableMetrics(ctx context.Context, db databaseName) (map[tableIdentifier]tableStats, error)
	getBlocksReadByTable(ctx context.Context, db databaseName) (map[tableIdentifier]tableIOStats, error)
```

Would this make sense?

Reply:
@djaglowski 

After playing around for a little bit, my opinion is that there would be a lot of conversion back and forth to the type alias and string especially i.e. iterating over the databases 
```go
	databases := p.config.Databases
        ...
	if len(databases) == 0 {
		dbList, err := listClient.listDatabases(ctx)
        ...
	}

	dbNames := []databaseName{}
	for _, db := range databases {
		dbNames = append(dbNames, databaseName(db))
	}
```

There is also something nice about it being separate from the databaseNames that generate the `where` clause on some of the queries so if we need to maybe refactor this component to have unique instance IDs per database when we record, then we wouldn't have to revisit the queries and separate it out.

Along with there being some complications mocks makes me hesitant to adopt this change. Happy to look into it more if you feel like it is preferred!

Reply:
That's fine. Not trying to make it complicated.